### PR TITLE
zero token operations

### PR DESCRIPTION
**Pull request by littletarzan**
Signed-off-by: littletarzan <mdelore.ufl@gmail.com>

**Description**:

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hip-0000-zero-token-operations.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+superseded-by: <HIP number(s)>
+---
+
+## Abstract
+
+Integer rounding will naturally lead to DeFi contracts sometimes minting or transferring 0 units of a token. But the network rejects 0 units in these operations, forcing contract developers to include special cases that would not be necessary on most networks.
+
+## Motivation
+
+To reduce code, if statement checks, etc inside smart contracts and make DeFi contracts easier to use. To reduce bugs in DeFi applications involving math and rounding.
```
Comment (mgarbs) [inline hip-0000-zero-token-operations.md:24]: Perhaps we can flesh this out a bit?
From [hip-1](https://hips.hedera.com/hip/hip-1):
The motivation should clearly explain why the existing specification is inadequate to address the problem that the HIP solves.

File: hip-0000-zero-token-operations.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+## Rationale
+
+DeFi contracts using ERC20 standard may mint, burn, and transfer zero tokens.
+
+The rationale should provide evidence of consensus within the community and discuss important objections or concerns raised during the discussion.
```
Comment (mgarbs) [inline hip-0000-zero-token-operations.md:30]: There is some default text here from the template hip that we should remove. Also, the rationale is your chance to really explain the reasoning behind why the hip is useful. From hip-1:

_The rationale fleshes out the specification by describing why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages._

_The rationale should provide evidence of consensus within the community and discuss important objections or concerns raised during the discussion._

**Review by mgarbs — CHANGES_REQUESTED** [review]
I left a few comments. Also, a couple of things:

- This hip should be inside the `/HIP` folder of your fork instead of the root directory.
- This hip is failing the validation check https://github.com/hashgraph/hedera-improvement-proposal/runs/8209056776?check_suite_focus=true You'll need to tweak the headers a bit in order for it to pass.

File: hip-0000-zero-token-operations.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+---
+hip: <HIP number (this is determined by the HIP editor)>
```
Comment (mgarbs) [inline hip-0000-zero-token-operations.md:2]:
```suggestion
hip: 0000
```

File: hip-0000-zero-token-operations.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+title: Zero unit token operations in smart contracts
+author: littletarzan, vae.vecturne
```
Comment (mgarbs) [inline hip-0000-zero-token-operations.md:4]: should be:
author: Firstname LastName <email>, Firstname1 LastName1 <email1>

File: hip-0000-zero-token-operations.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+working-group:
+type: Standards track
+category: Service
+needs-council-approval: No
+status: Draft
+created: Sept 1, 2022
```
Comment (mgarbs) [inline hip-0000-zero-token-operations.md:10]:
```suggestion
created: 2022-09-01
```

File: hip-0000-zero-token-operations.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/563
+updated: <comma separated list of dates>
+requires: <HIP number(s)>
+replaces: <HIP number(s)>
+superseded-by: <HIP number(s)>
```
Comment (mgarbs) [inline hip-0000-zero-token-operations.md:15]:
```suggestion
```

File: hip-0000-zero-token-operations.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
+---
+hip: <HIP number (this is determined by the HIP editor)>
+title: Zero unit token operations in smart contracts
+author: littletarzan, vae.vecturne
+working-group:
+type: Standards track
+category: Service
+needs-council-approval: No
```
Comment (mgarbs) [inline hip-0000-zero-token-operations.md:8]:
```suggestion
needs-council-approval: Yes
```

**Comment by littletarzan** [comment]
@mgarbs thank you, I made the requested changes

File: HIP/hip-0000-zero-token-operations.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+## Motivation
+
+To reduce code, if statement checks, etc inside smart contracts and make DeFi contracts easier to use. To reduce bugs in DeFi applications involving math and rounding.
+
+## Rationale
+
+DeFi contracts using ERC20 standard may mint, burn, and transfer zero tokens.
+
+The rationale should provide evidence of consensus within the community and discuss important objections or concerns raised during the discussion.
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:26]:
```suggestion
```

File: HIP/hip-0000-zero-token-operations.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/563
+---
+
+## Abstract
+
+Integer rounding will naturally lead to DeFi contracts sometimes minting or transferring 0 units of a token. But the network rejects 0 units in these operations, forcing contract developers to include special cases that would not be necessary on most networks.
+
+## Motivation
+
+To reduce code, if statement checks, etc inside smart contracts and make DeFi contracts easier to use. To reduce bugs in DeFi applications involving math and rounding.
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:20]:
```suggestion
Currently, whenever a zero value parameter passes through a mint, burn or transfer transaction, the network rejects it immediately. Frequently this happens with integer rounding calculations in smart contracts. To get around it, contract developers must check for a non-zero value to submit mint, burn or transfer transactions, which is inconsistent with most other networks. This hip will eliminate `if` statements checking for zero before mint, burn or transfer transactions in smart contracts making them easier to use and be in alignment with most other networks.
```

File: HIP/hip-0000-zero-token-operations.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+The response code for token mint, burn, and transfer returns SUCCESS = 22
+
+## Backwards Compatibility
+
+This would be backward compatible.
+
+## Security Implications
+
+
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:46]:
```suggestion
N/A
```

File: HIP/hip-0000-zero-token-operations.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+
+## How to Teach This
+
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:49]:
```suggestion
N/A
```

File: HIP/hip-0000-zero-token-operations.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+
+## Reference Implementation
+
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:53]:
```suggestion
#### Example :

Example of a transfer transaction that should be allowed if a 0 amount is passed.
```js
function mint(uint amount) external {
        balanceOf[msg.sender] += amount;
        totalSupply += amount;
        emit Transfer(address(0), msg.sender, amount);
 }
 
```
```

**Review by mgarbs — CHANGES_REQUESTED** [review]

File: HIP/hip-0000-zero-token-operations.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
+---
+hip: 0000
+title: Zero unit token operations in smart contracts
+author: Matthew DeLorenzo <mdelore.ufl@gmail.com>, Vae Vecturne <vea.vecturne@gmail.com>
+working-group:
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:5]:
```suggestion
```

File: HIP/hip-0000-zero-token-operations.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+## Reference Implementation
+#### Example :
+Example of a transfer transaction that should be allowed if a 0 amount is passed.
+```js
+function mint(uint amount) external {
+        balanceOf[msg.sender] += amount;
+        totalSupply += amount;
+        emit Transfer(address(0), msg.sender, amount);
+ }
+
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:58]:
```suggestion
```
```

File: HIP/hip-0000-zero-token-operations.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+N/A
+
+## How to Teach This
+
+N/A
+
+## Reference Implementation
+#### Example :
+Example of a transfer transaction that should be allowed if a 0 amount is passed.
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:51]:
```suggestion
Example of a mint transaction that should be allowed if a 0 amount is passed.
```

File: HIP/hip-0000-zero-token-operations.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
+---
+hip: 0000
+title: Zero unit token operations in smart contracts
+author: Matthew DeLorenzo <mdelore.ufl@gmail.com>, Vae Vecturne <vea.vecturne@gmail.com>
+type: Standards track
```
Comment (mgarbs) [inline HIP/hip-0000-zero-token-operations.md:5]:
```suggestion
type: Standards Track
```

**Review by mgarbs — DISMISSED** [review]

**Review by mgarbs — APPROVED** [review]

**Pull request merged**